connector/src/main/java/org/geysermc/connector/network/translators/collision/CollisionManager.java

```diff
@@ -342,40 +322,37 @@ private Vector3d correctMovementForCollisions(Vector3d movement, BoundingBox bou
         BoundingBox movementBoundingBox = boundingBox.clone();
         movementBoundingBox.extend(movement);
 
-        List<Vector3i> collidableBlocks = getCollidableBlocks(movementBoundingBox);
-
         if (Math.abs(movementY) > CollisionManager.COLLISION_TOLERANCE) {
-            movementY = computeCollisionOffset(boundingBox, Axis.Y, movementY, collidableBlocks, checkWorld);
+            movementY = computeCollisionOffset(boundingBox, Axis.Y, movementY, movementBoundingBox, checkWorld);
             boundingBox.translate(0, movementY, 0);
         }
         boolean checkZFirst = Math.abs(movementZ) > Math.abs(movementX);
         if (checkZFirst && Math.abs(movementZ) > CollisionManager.COLLISION_TOLERANCE) {
-            movementZ = computeCollisionOffset(boundingBox, Axis.Z, movementZ, collidableBlocks, checkWorld);
+            movementZ = computeCollisionOffset(boundingBox, Axis.Z, movementZ, movementBoundingBox, checkWorld);
             boundingBox.translate(0, 0, movementZ);
         }
         if (Math.abs(movementX) > CollisionManager.COLLISION_TOLERANCE) {
-            movementX = computeCollisionOffset(boundingBox, Axis.X, movementX, collidableBlocks, checkWorld);
+            movementX = computeCollisionOffset(boundingBox, Axis.X, movementX, movementBoundingBox, checkWorld);
             boundingBox.translate(movementX, 0, 0);
         }
         if (!checkZFirst && Math.abs(movementZ) > CollisionManager.COLLISION_TOLERANCE) {
-            movementZ = computeCollisionOffset(boundingBox, Axis.Z, movementZ, collidableBlocks, checkWorld);
+            movementZ = computeCollisionOffset(boundingBox, Axis.Z, movementZ, movementBoundingBox, checkWorld);
             boundingBox.translate(0, 0, movementZ);
         }
 
         boundingBox.translate(-movementX, -movementY, -movementZ);
         return Vector3d.from(movementX, movementY, movementZ);
     }
 
-    private double computeCollisionOffset(BoundingBox boundingBox, Axis axis, double offset, List<Vector3i> collidableBlocks, boolean checkWorld) {
-        for (Vector3i blockPos : collidableBlocks) {
+    private double computeCollisionOffset(BoundingBox boundingBox, Axis axis, double offset, BoundingBox movementBoundingBox, boolean checkWorld) {
+        for (BlockPositionIterator iter = collidableBlocksIterator(movementBoundingBox); iter.hasNext(); iter.next()) {
             if (checkWorld) {
-                BlockCollision blockCollision = BlockUtils.getCollisionAt(session, blockPos);
+                BlockCollision blockCollision = BlockUtils.getCollisionAt(session, iter.getX(), iter.getY(), iter.getZ());
```

Comment:
Since getX/getY/getZ is potentially used twice, you can pull them out into local variables above the if statement here.

Reply:
Is there a point to storing the result of getX/getY/getZ here? There are duplicated calls in other places.

Reply:
You use the values here, on line 353, and line 356. There's a small calculation to find each value, so I feel it would be cleaner to just save those values for the for loop iteration.

Reply:
If the goal is to avoid doing the calculation more than once, wouldn't it be better to convert getX/getY/getZ into regular getters and do the calculations in next?

Reply:
We only need to use a value multiple times here, so I would personally say no.